### PR TITLE
Fix duplicate env vars in container

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
Currently, when building a k8s Pod we make duplicate calls to `AddFlyteCustomizationsToContainer` ([here](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/container_helper.go#L260) and [here](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/pod_helper.go#L207). This PR updates the [BuildRawPod function](https://github.com/flyteorg/flyteplugins/blob/aa1ed678ec085a78f3295e51e6153fb56e9bf1ed/go/tasks/pluginmachinery/flytek8s/pod_helper.go#L120) to call [a refactored BuildRawContainer](https://github.com/flyteorg/flyteplugins/blob/aa1ed678ec085a78f3295e51e6153fb56e9bf1ed/go/tasks/pluginmachinery/flytek8s/container_helper.go#L199) to ensure the `AddFlyteCustomizationsToContainer` is only called once so that env vars are only injected a single time.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 ^^^

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/3740

## Follow-up issue
_NA_

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]

**Review by eapolinario — COMMENTED** [review]
worth adding a couple of unit tests to confirm behavior?

**Review by pingsutw — APPROVED** [review]
